File: core/src/main/java/im/mingxi/miko/startup/util/XRes.java
package im.mingxi.miko.startup.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.res.AssetManager;
import android.content.res.Resources;
import im.mingxi.loader.util.PathUtil;
import java.lang.reflect.Method;

public abstract class XRes {

  public static void addAssetsPath(Context context) {
    addAssetsPath(context.getResources(), PathUtil.moduleApkPath);
  }
    
    public static void addAssetsPath(Context context, String path) {
    addAssetsPath(context.getResources(), path);
  }

  @SuppressLint({"PrivateApi", "DiscouragedPrivateApi"})
  public static void addAssetsPath(Resources resources, String str) {
    try {
      Method method = AssetManager.class.getDeclaredMethod("addAssetPath", String.class);
      method.setAccessible(true);
      method.invoke(resources.getAssets(), str);
    } catch (Exception unused) {
    }
  }
}

File: core/src/main/java/im/mingxi/miko/startup/util/XRes.kt
package im.mingxi.miko.startup.util

import android.annotation.SuppressLint
import android.content.Context
import android.content.res.AssetManager
import android.content.res.Resources
import im.mingxi.loader.util.PathUtil

object XRes {
    @JvmStatic
    fun addAssetsPath(context: Context) {
        addAssetsPath(context.resources, PathUtil.moduleApkPath)
    }

    @JvmStatic
    fun addAssetsPath(context: Context, path: String?) {
        addAssetsPath(context.resources, path)
    }

    @SuppressLint("PrivateApi", "DiscouragedPrivateApi")
    @JvmStatic
    @Throws(Exception::class)
    fun addAssetsPath(resources: Resources, str: String?) {
        val method =
            AssetManager::class.java.getDeclaredMethod("addAssetPath", String::class.java)
        method.isAccessible = true
        method.invoke(resources.assets, str)
    }
}